iroh-resolver/src/balanced_tree.rs

```diff
@@ -0,0 +1,388 @@
+use std::collections::VecDeque;
+
+use anyhow::Result;
+use async_stream::try_stream;
+use bytes::{Bytes, BytesMut};
+use cid::Cid;
+use futures::{Stream, StreamExt};
+
+use crate::unixfs::{dag_pb, unixfs_pb, DataType, Node, UnixfsNode};
+use crate::unixfs_builder::encode_unixfs_pb;
+
+// 3                          root
+// 2                                 nextparent0
+// 1             parent0              parent1    ... parent7
+// 0 [0, 1, 2, 3, 4, 5, 6, 7]      [0, ... ]
+//
```

Comment:
we can probably remove our ascii art at some point 😅

Reply:
I'm conceding to moving some of it, lol
Some I think will be helpful for anyone reading it over in the future